src/hotspot/os_cpu/linux_aarch64/vm_version_linux_aarch64.cpp

```diff
@@ -185,7 +185,10 @@ static bool read_fully(const char *fname, char *buf, size_t buflen) {
   assert(buflen >= 1, "invalid argument");
   int fd = os::open(fname, O_RDONLY, 0);
   if (fd != -1) {
+    PRAGMA_DIAG_PUSH
+    PRAGMA_NONNULL_IGNORED
     ssize_t read_sz = ::read(fd, buf, buflen);
+    PRAGMA_DIAG_POP
```

Comment:
Does hardening `buf != nullptr` to a `guarantee()` fix it?

Reply:
Thanks, use guarantee instead of assert can suppress the gcc warnings. I think it's better than use suppression pragma macro.